### PR TITLE
Fix hsl,hsla,rgb,rba functions to use comma and commaless syntax

### PR DESCRIPTION
**Pull request by Elchi3**
See https://github.com/mdn/data/pull/39#issuecomment-276453778

### CODE REVIEW

File: css/syntaxes.json [hunk 1]
```diff
@@ -138,9 +138,8 @@
   "repeat-style": "repeat-x | repeat-y | [ repeat | space | round | no-repeat ]{1,2}",
   "repeating-linear-gradient()": "repeating-linear-gradient( [ &lt;angle&gt; | to &lt;side-or-corner&gt; ]? , &lt;color-stop-list&gt; )",
   "repeating-radial-gradient()": "repeating-radial-gradient( [ &lt;ending-shape&gt; || &lt;size&gt; ]? [ at &lt;position&gt; ]? , &lt;color-stop-list&gt; )",
-  "rgb()": "rgb( &lt;rgb-component&gt;#{3} , &lt;alpha-value&gt;? )",
-  "rgb-component": "&lt;integer&gt; | &lt;percentage&gt;",
-  "rgba()": "rgba( &lt;rgb-component&gt;#{3} , &lt;alpha-value&gt; )",
+  "rgb()": "rgb( [[ &lt;percentage&gt;{3} | &lt;number&gt;{3} ] [ / &lt;alpha-value&gt; ]? ] | [[ &lt;percentage&gt;#{3} | &lt;number&gt;#{3}] , &lt;alpha-value&gt;? ] )",
+  "rgba()":"rgba [[ &lt;percentage&gt;{3} | &lt;number&gt;{3} ] [ / &lt;alpha-value&gt; ]? ] | [[ &lt;percentage&gt;#{3} | &lt;number&gt;#{3}] , &lt;alpha-value&gt;? ] )",
```
Comment (SebastianZ) [inline css/syntaxes.json]: The opening bracket is missing.

Sebastian

**Comment by Elchi3** [comment]
Updated.

File: css/syntaxes.json [hunk 2]
```diff
@@ -138,9 +138,8 @@
   "repeat-style": "repeat-x | repeat-y | [ repeat | space | round | no-repeat ]{1,2}",
   "repeating-linear-gradient()": "repeating-linear-gradient( [ &lt;angle&gt; | to &lt;side-or-corner&gt; ]? , &lt;color-stop-list&gt; )",
   "repeating-radial-gradient()": "repeating-radial-gradient( [ &lt;ending-shape&gt; || &lt;size&gt; ]? [ at &lt;position&gt; ]? , &lt;color-stop-list&gt; )",
-  "rgb()": "rgb( &lt;rgb-component&gt;#{3} , &lt;alpha-value&gt;? )",
-  "rgb-component": "&lt;integer&gt; | &lt;percentage&gt;",
-  "rgba()": "rgba( &lt;rgb-component&gt;#{3} , &lt;alpha-value&gt; )",
+  "rgb()": "rgb( [ [ &lt;percentage&gt;{3} | &lt;number&gt;{3} ] [ / &lt;alpha-value&gt; ]? ] | [ [ &lt;percentage&gt;#{3} | &lt;number&gt;#{3}] , &lt;alpha-value&gt;? ] )",
+  "rgba()":"rgba( [ [ &lt;percentage&gt;{3} | &lt;number&gt;{3} ] [ / &lt;alpha-value&gt; ]? ] | [ [ &lt;percentage&gt;#{3} | &lt;number&gt;#{3}] , &lt;alpha-value&gt;? ] )",
```
Comment (lahmatiy) [inline css/syntaxes.json]: Space after colon missed

Comment (Elchi3) [inline css/syntaxes.json]: Thanks @lahmatiy! Fixed now.

**Comment by Elchi3** [comment]
Can you have another look and merge if this is OK, @SebastianZ?

File: css/syntaxes.json [hunk 3]
```diff
@@ -138,9 +138,8 @@
   "repeat-style": "repeat-x | repeat-y | [ repeat | space | round | no-repeat ]{1,2}",
   "repeating-linear-gradient()": "repeating-linear-gradient( [ &lt;angle&gt; | to &lt;side-or-corner&gt; ]? , &lt;color-stop-list&gt; )",
   "repeating-radial-gradient()": "repeating-radial-gradient( [ &lt;ending-shape&gt; || &lt;size&gt; ]? [ at &lt;position&gt; ]? , &lt;color-stop-list&gt; )",
-  "rgb()": "rgb( &lt;rgb-component&gt;#{3} , &lt;alpha-value&gt;? )",
-  "rgb-component": "&lt;integer&gt; | &lt;percentage&gt;",
-  "rgba()": "rgba( &lt;rgb-component&gt;#{3} , &lt;alpha-value&gt; )",
+  "rgb()": "rgb( [ [ &lt;percentage&gt;{3} | &lt;number&gt;{3} ] [ / &lt;alpha-value&gt; ]? ] | [ [ &lt;percentage&gt;#{3} | &lt;number&gt;#{3}] , &lt;alpha-value&gt;? ] )",
```
Comment (lahmatiy) [inline css/syntaxes.json]: One more missed space `&lt;number&gt;#{3}]` -> `&lt;number&gt;#{3} ]`

**Comment by lahmatiy** [comment]
@Elchi3 Are you sure it's time to add new color syntax? Spec in draft status and no support by any browser yet (or I miss something?).

**Comment by Elchi3** [comment]
@lahmatiy if I see it correctly, this was implemented in https://bugzilla.mozilla.org/show_bug.cgi?id=1295456 which is in Firefox 52, which ships in 2 weeks (March 7).

File: css/syntaxes.json (hunk 3 above)
Comment (Elchi3) [inline css/syntaxes.json]: Gah! Thanks, fixed!

**Comment by SebastianZ** [comment]
> Can you have another look and merge if this is OK, @SebastianZ?

Looks good to me. Merged. Thanks!

Sebastian

**Pull request merged**